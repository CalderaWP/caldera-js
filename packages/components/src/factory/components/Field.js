import React, { Fragment } from 'react';
import { fieldFactory } from '../factories/fieldFactory';

export const Field = ({ field, onChange, onBlur, messages }) => (
	<Fragment>{fieldFactory(field, onChange, onBlur, messages)}</Fragment>
);
