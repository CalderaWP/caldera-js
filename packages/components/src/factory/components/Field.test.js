import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	radioField,
	emailField
} from '../fields.fixtures';

import { Field } from './Field';

describe('Field component', () => {
	let onChange;
	let onBlur;
	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	it('Creates a text field', () => {
		const component = renderer.create(
			<Field field={textField} onChange={onChange} onBlur={onBlur} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
	it('Creates a number field', () => {
		const component = renderer.create(
			<Field field={numberField} onChange={onChange} onBlur={onBlur} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Creates an email field', () => {
		const component = renderer.create(
			<Field field={emailField} onChange={onChange} onBlur={onBlur} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Creates an checkbox field field', () => {
		const component = mount(
			<Field field={checkboxField} onChange={onChange} onBlur={onBlur} />
		);
		expect(component.find('input').prop('type')).toEqual('checkbox');
	});

	it('Creates an checkbox field set', () => {
		const component = renderer.create(
			<Field
				field={checkboxFieldset}
				onChange={onChange}
				onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Creates an select field', () => {
		const component = renderer.create(
			<Field field={selectField} onChange={onChange} onBlur={onBlur} />
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Creates an select field identified as dropdown', () => {
		const component = renderer.create(
			<Field
				field={{
					...selectField,
					fieldType: 'dropdown'
				}}
				onChange={onChange}
				onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Changes calls change handler of text field', () => {
		const component = mount(
			<Field field={textField} onChange={onChange} onBlur={onBlur} />
		);

		component.find('input').simulate('change');
		expect(onChange.mock.calls.length).toBe(1);
	});
	it('Changes calls change handler of select field', () => {
		const component = mount(
			<Field field={selectField} onChange={onChange} onBlur={onBlur} />
		);

		component.find('select').simulate('change');
		expect(onChange.mock.calls.length).toBe(1);
	});
	it('Changes calls change handler of radio field', () => {
		const component = mount(
			<Field field={radioField} onChange={onChange} onBlur={onBlur} />
		);

		component
			.find('input')
			.first()
			.simulate('change');
		expect(onChange.mock.calls.length).toBe(1);
	});
	it('Changes calls change handler of checkbox field', () => {
		const component = mount(
			<Field field={checkboxField} onChange={onChange} onBlur={onBlur} />
		);

		component.find('input').simulate('change');
		expect(onChange.mock.calls.length).toBe(1);
	});
});
