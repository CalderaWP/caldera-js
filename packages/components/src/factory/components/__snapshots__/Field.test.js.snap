// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component Creates a number field 1`] = `
<div
  className="components-base-control caldera-field caldera-field-number"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-1"
    >
      Number Field Label
    </label>
    <input
      aria-describedby="inspector-text-control-1__help"
      className="components-text-control__input"
      id="numberFieldId"
      max={10}
      min={1}
      onBlur={[Function]}
      onChange={[Function]}
      placeholder={undefined}
      type="number"
      value={3}
    />
  </div>
  <p
    className="components-base-control__help"
    id="inspector-text-control-1__help"
  >
    Number field description
  </p>
</div>
`;

exports[`Field component Creates a text field 1`] = `
<div
  className="components-base-control caldera-field caldera-field-text"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-0"
    >
      First Name
    </label>
    <input
      aria-describedby="inspector-text-control-0__help"
      className="components-text-control__input"
      id="firstName"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder={undefined}
      type="text"
      value="roy"
    />
  </div>
  <p
    className="components-base-control__help"
    id="inspector-text-control-0__help"
  >
    your first name
  </p>
</div>
`;

exports[`Field component Creates an checkbox field set 1`] = `
<fieldset
  className="caldera-fieldset caldera-fieldset-checkboxes"
>
  <legend>
    Checkbox Field Set Label
  </legend>
  <div
    className="components-base-control"
  >
    <div
      className="components-base-control__field"
    >
      <input
        aria-describedby={undefined}
        checked={true}
        className="components-checkbox-control__input"
        id="opt-1"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value="1"
      />
      <label
        className="components-checkbox-control__label"
        htmlFor="inspector-checkbox-control-1"
      >
        One
      </label>
    </div>
  </div>
  <div
    className="components-base-control"
  >
    <div
      className="components-base-control__field"
    >
      <input
        aria-describedby="inspector-checkbox-control-2__help"
        checked={true}
        className="components-checkbox-control__input"
        id="opt-2"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value="1"
      />
      <label
        className="components-checkbox-control__label"
        htmlFor="inspector-checkbox-control-2"
      >
        Two
      </label>
    </div>
    <p
      className="components-base-control__help"
      id="inspector-checkbox-control-2__help"
    >
      The Second Option
    </p>
  </div>
  <div
    className="components-base-control"
  >
    <div
      className="components-base-control__field"
    >
      <input
        aria-describedby={undefined}
        checked={true}
        className="components-checkbox-control__input"
        id="opt-3"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value="1"
      />
      <label
        className="components-checkbox-control__label"
        htmlFor="inspector-checkbox-control-3"
      >
        Three
      </label>
    </div>
  </div>
</fieldset>
`;

exports[`Field component Creates an email field 1`] = `
<div
  className="components-base-control caldera-field caldera-field-email"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-2"
    >
      Email Field Label
    </label>
    <input
      aria-describedby="inspector-text-control-2__help"
      className="components-text-control__input"
      id="emailFieldId"
      multiple={true}
      onBlur={[Function]}
      onChange={[Function]}
      placeholder={undefined}
      type="email"
      value="emai@email.com"
    />
  </div>
  <p
    className="components-base-control__help"
    id="inspector-text-control-2__help"
  >
    Email field description
  </p>
</div>
`;

exports[`Field component Creates an select field 1`] = `
<div
  className="components-base-control caldera-field caldera-field-select"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-select-control-0"
    >
      Select Field Label
    </label>
    <select
      aria-describedby="inspector-select-control-0__help"
      className="components-select-control__input"
      id="selectFieldId"
      multiple={false}
      onChange={[Function]}
      value={1}
    >
      <option
        value={0}
      >
        Zero
      </option>
      <option
        value={1}
      >
        One
      </option>
      <option
        value={2}
      >
        Two
      </option>
    </select>
  </div>
  <p
    className="components-base-control__help"
    id="inspector-select-control-0__help"
  >
    Select field description
  </p>
</div>
`;

exports[`Field component Creates an select field identified as dropdown 1`] = `
<div
  className="components-base-control caldera-field caldera-field-select"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-select-control-1"
    >
      Select Field Label
    </label>
    <select
      aria-describedby="inspector-select-control-1__help"
      className="components-select-control__input"
      id="selectFieldId"
      multiple={false}
      onChange={[Function]}
      value={1}
    >
      <option
        value={0}
      >
        Zero
      </option>
      <option
        value={1}
      >
        One
      </option>
      <option
        value={2}
      >
        Two
      </option>
    </select>
  </div>
  <p
    className="components-base-control__help"
    id="inspector-select-control-1__help"
  >
    Select field description
  </p>
</div>
`;
